lib/pure/ospaths.nim

```diff
@@ -383,6 +383,13 @@ when not declared(getEnv) or defined(nimscript):
     else:
       result = splitPath(path).tail
 
+  proc extractExt*(path: string): string {.
```

Comment:
This procedure is not necessary. The ``os`` module defines a ``splitFile`` procedure which splits a filename for you into ``dir``, ``name`` and ``ext``, you can simply write ``splitFile(path).ext``.